src/Nethermind/Nethermind.Evm/StateOverridesExtensions.cs

```diff
@@ -71,13 +69,15 @@ void ApplyState(Dictionary<UInt256, Hash256> diff)
 
     private static void UpdateCode(
         this IWorldState stateProvider,
-        OverridableCodeInfoRepository overridableCodeInfoRepository,
+        IOverridableCodeInfoRepository overridableCodeInfoRepository,
         IReleaseSpec currentSpec,
         AccountOverride accountOverride,
         Address address)
     {
         if (accountOverride.Code is not null)
         {
+            stateProvider.InsertCode(address, accountOverride.Code, currentSpec);
```

Comment:
This is needed to prevent an "Account {address} is null when updating storage hash" error in `StateProvider` when overridden code tries to modify address storage.

Reply:
@OlegJakushkin maybe this solves some issues on StateRoot in eth_simulate?